app/controllers/blogs_controller.rb

```diff
@@ -0,0 +1,97 @@
+class BlogsController < ApplicationController
+  before_action :set_blog, only: [:show, :edit, :update, :destroy, :posts]
+  before_action :authenticate_user!, except: [:index, :posts, :post_details]
+
+  # GET /blogs
+  # GET /blogs.json
+  def index
+    @blogs = Blog.all
+  end
+
+  def posts
+    if params[:q_post].present?
+      @posts = post_search(@blog, params[:q_post])
+    else
+      @posts = @blog.posts.published
+    end
+  end
+
+  def post_details
+    @post = Post.friendly.find(params[:post_id])
+    commontator_thread_show(@post)
+  end
+
+  def manage
+    @blogs = Blog.where(user_id: current_user.id)
+  end
+
+  # GET /blogs/1
+  # GET /blogs/1.json
+  def show
+  end
+
+  # GET /blogs/new
+  def new
+    @blog = Blog.new
+  end
+
+  # GET /blogs/1/edit
+  def edit
+  end
+
+  # POST /blogs
+  # POST /blogs.json
+  def create
+    @blog = Blog.new(blog_params)
+    @blog.user_id = current_user.id
+
+    respond_to do |format|
+      if @blog.save
+        format.html { redirect_to @blog, notice: 'Blog was successfully created.' }
+        format.json { render :show, status: :created, location: @blog }
+      else
+        format.html { render :new }
+        format.json { render json: @blog.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /blogs/1
+  # PATCH/PUT /blogs/1.json
+  def update
+    respond_to do |format|
+      if @blog.update(blog_params)
+        format.html { redirect_to @blog, notice: 'Blog was successfully updated.' }
+        format.json { render :show, status: :ok, location: @blog }
+      else
+        format.html { render :edit }
+        format.json { render json: @blog.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /blogs/1
+  # DELETE /blogs/1.json
+  def destroy
+    @blog.destroy
+    respond_to do |format|
+      format.html { redirect_to blogs_url, notice: 'Blog was successfully destroyed.' }
+      format.json { head :no_content }
+    end
+  end
+
+  private
+    # Search posts by title, body and tags
+    def post_search(blog, term)
+      (blog.posts.published.contains_text(term).tagged_with(term) + blog.posts.published.tagged_with(term)).uniq
```

Comment:
I tried searching for "Improve" at http://localhost:3000/blogs/the-little-magpie/posts, and it didn't find the first (and only) article as I'd have expected. Any thoughts on why that might be? 

I also have some concerns about this implementation from a performance perspective. Since it's only searching a single blog it's probably not a big deal, but if a blog had, say, 100k posts and 50k of them happen to have a "t" in the title, and someone searches for "t", this would be problematic, as you'd be loading >50k records into memory then doing a `uniq` on them. It's a pretty resource-intensive solution for something that could easily be pushed to the database!

Reply:
The search definitely had some issues, its looking better now; also, I've used raw query to so we're not loading records in memory; 
Further to this, we can also add limit check (e.g. at least 4 characters) on the search parameter (I havn't added it yet but happy to do it if required)